akka-actor-typed/src/main/scala/akka/actor/typed/TypedActorContext.scala

```diff
@@ -25,4 +27,19 @@ trait TypedActorContext[T] {
    * Get the `scaladsl` of this `ActorContext`.
    */
   def asScala: scaladsl.ActorContext[T]
+
+  /**
+   * Narrow the type of this `TypedActorContext`, which is always a safe operation.
+   */
+  def narrow[U <: T]: TypedActorContext[U]
```

Comment:
that seems safe but when would it be needed?

Reply:
if we would add it then it should be in scaladsl.ActorContext and javadsl.ActorContext so that it would keep the specific return type